livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterprereter.java

```diff
@@ -168,18 +168,16 @@ private String extractStatementResult(String result) {
   @Override
   public void cancel(InterpreterContext context) {
     if (livyVersion.isCancelSupported()) {
+      String paraId = context.getParagraphId();
```

Comment:
yap, could paraId be NULL? If it could we should check since ConcurrentHashMap throws if it is